index.bs

```diff
@@ -575,79 +575,225 @@ urlPrefix: https://w3c.github.io/FileAPI/#dfn-; type: dfn;
+
+		A {{ClipboardItem}} object has an associated <dfn for="ClipboardItem">clipboard item</dfn>, which is a [=clipboard item=].
+
+		A {{ClipboardItem}} object has an associated <dfn for="ClipboardItem">types array</dfn>, which is a {{FrozenArray}}.
+
+		To <dfn>create a {{ClipboardItem}} object</dfn>, given a [=clipboard item=] |clipboardItem|'s [=relevant realm=] |realm|, run these steps:
+			1. Let |clipboardItemObject| be a [=new=] {{ClipboardItem}} with |realm|.
+
+			1. Set |clipboardItemObject|'s [=clipboard item=] to |clipboardItem|.
+
+		The <a constructor lt="ClipboardItem()">constructor</a> steps for <code>new ClipboardItem(<var>items</var>, <var>options</var>)</code> are:
+			1. Set [=this=]'s [=ClipboardItem/clipboard item=] to a new [=clipboard item=].
+
+			1. Set [=this=]'s [=ClipboardItem/clipboard item=]'s [=presentation style=] to |options|["{{ClipboardItemOptions/presentationStyle}}"].
+
+			1. For each (|key|, |value|) in |items|:
+
+				1. Let |mimeType| be the result of [=parse a mime type=] given |key|.
+
+				1. If |mimeType| is failure, then throw a {{TypeError}}.
+
+				1. Let |representation| be a new [=representation=].
+
+				1. Set |representation|'s [=mime type=] to |key|.
+
+				1. Set |representation|'s [=data=] to |value|.
+
+				1. Append |representation| to [=this=]'s [=ClipboardItem/clipboard item=]'s [=list of representations=].
+
+				1. Let |types| be the result of running [=create a frozen array=] of {{DOMString}}.
+
+				1. Let |mimeTypeString| be the result of [=serialize a mime type=] with |mimeType|.
+				
+				1. Add |mimeTypeString| to |types|.
+
+				1. Set |types| to [=this=]'s [=ClipboardItem/types array=].
+
+		<h4 attribute for=ClipboardItem lt=presentationStyle>presentationStyle</h4>
+		<p>
+			The {{ClipboardItem/presentationStyle}} getter steps are to return [=this=]'s [=ClipboardItem/clipboard item=]'s [=presentation style=].
+		</p>
+
+		<h4 attribute for=ClipboardItem lt=types>types</h4>
+		<p>
+			{{ClipboardItem/types}} getter steps are to return [=this=]'s [=ClipboardItem/types array=].
+		</p>
+
+		<h4 method for=ClipboardItem lt=getType(type)>getType(type)</a> must run the below steps:</h4>
+
+			1. Let |realm| be [=this=]'s [=relevant realm=].
+
+			1. Let |mimeType| be the result of [=parse a mime type=] given |type|.
+
+			1. If |mimeType| is failure, then throw a {{TypeError}}.
 
-		enum PresentationStyle { "unspecified", "inline", "attachment" };
+			1. Let |itemTypeList| be [=this=]'s [=ClipboardItem/clipboard item=]'s [=list of representations=].
 
-		dictionary ClipboardItemOptions {
-		  PresentationStyle presentationStyle = "unspecified";
+			1. For each |representation| in |itemTypeList|:
+
+				1. If |representation|'s [=mime type=] is |mimeType|, then:
+
+					1. Let |blobData| be a [=Blob=] created using |representation|'s [=data=].
```

Comment:
Do you want to set the Blob's type as well? It looks like the File API spec is not very precise in a way that would make this rigorous, but I would guess you might want to do something like the following:

```
1. Let |blobData| be a [=Blob=] created using |representation|'s [=data=] and with its type set to |mimeType|, [=serialize a MIME type|serialized=].
```

Reply:
makes sense. Added this step.